### PR TITLE
add krb5_conf_d_path & krb5_conf_d_purge params

### PR DESCRIPTION
**Pull request by jhoblitt**

### CODE REVIEW

**Comment by riton** [comment]
Hi @jhoblitt 
Thanks for the contribution. LGTM.

I just need to try this on our KDCs to check if nothing breaks.

**Comment by jhoblitt** [comment]
@riton I have it in production on a few hundred nodes without an incident.

It would be great if this mod adopted rspec-beaker.

File: manifests/init.pp [hunk 1]
```diff
@@ -291,7 +291,9 @@
   Hash $capaths                     = {},
   Hash $appdefaults                 = {},
   Hash $realms                      = {},
-  Hash $dbmodules                   = {}
+  Hash $dbmodules                   = {},
+  String[1] $krb5_conf_d_path       = '/etc/krb5.conf.d',
```
Comment (riton) [inline manifests/init.pp]: Data type `Stdlib::Absolutepath` may be more appropriate ?

File: manifests/init.pp [hunk 2]
```diff
@@ -339,6 +341,16 @@
   create_resources('mit_krb5::realm', $realms)
   create_resources('mit_krb5::dbmodules', $dbmodules)
 
+  file { $krb5_conf_d_path:
+    ensure  => directory,
+    owner   => $krb5_conf_owner,
+    group   => $krb5_conf_group,
+    mode    => $krb5_conf_mode,
```
Comment (riton) [inline manifests/init.pp]: On a _RockyLinux 9_ host, default directory mode is `0755`

```
[root@67923e1aed89 /]# ls -laF /etc/krb5.conf.d -d
drwxr-xr-x 1 root root 4096 Jun 22 20:24 /etc/krb5.conf.d/

[root@67923e1aed89 /]# cat /etc/redhat-release 
Rocky Linux release 9.2 (Blue Onyx)
```

This seems like a good default value to keep.

What about creating a new param `Stdlib::Filemode $krb5_conf_d_mode = "0755"` ?

For note, `Stdlib::Filemode` was first introduced in `puppetlabs-stdlib` `4.21.0`.

**Review by riton — CHANGES_REQUESTED** [review]
Just two points to clarify about _Data Type_ and _default value_.

**Comment by riton** [comment]
> It would be great if this mod adopted rspec-beaker.

#33 was created to track this (very good) suggestion

File: manifests/init.pp (hunk 2 above)
Comment (jhoblitt) [reply manifests/init.pp]: I don't have a need to change the mode of the conf dir, so it seems reasonable to hardcode the mode.

**Comment by jhoblitt** [comment]
Revised based on review comments.

**Comment by riton** [comment]
Sorry again for the latency and thanks for the patience.

Thanks again for your contribution. I'll include that in our next release. :rocket:

**Pull request merged**